bn-apps/memberships-management/membership-management-demo/src/main/kotlin/com/r3/businessnetworks/memberships/demo/flows/Flows.kt

```diff
@@ -0,0 +1,124 @@
+import com.r3.businessnetworks.membership.flows.member.MembershipsListResponse
+import com.r3.businessnetworks.membership.flows.member.RequestMembershipFlow
+import com.r3.businessnetworks.membership.flows.member.service.MembershipsCacheHolder
+import com.r3.businessnetworks.membership.states.MembershipContract
+import com.r3.businessnetworks.membership.states.MembershipState
+
+
+import com.r3.businessnetworks.memberships.demo.contracts.SampleContract
+import com.r3.businessnetworks.memberships.demo.contracts.SampleState
+import net.corda.core.contracts.ReferencedStateAndRef
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.contracts.requireThat
+import net.corda.core.flows.*
+import net.corda.core.identity.Party
+import net.corda.core.serialization.CordaSerializable
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+import net.corda.core.utilities.unwrap
+
+@CordaSerializable
+class RequestMembershipForTransaction (val bno : Party)
+
+class IssueAssetFlow (val bno : Party) : FlowLogic<SignedTransaction>() {
+    @Suspendable
+    override fun call() : SignedTransaction {
+        // for the sake of demo we are just taking the first available notary
+        val notary = serviceHub.networkMapCache.notaryIdentities.single()
+
+        // query the vault with bno
+        val memberships= subFlow(GetMembershipsFlow(bno))
+        val ourMembership = memberships[ourIdentity]!!
+
+        val builder = TransactionBuilder(notary)
+                .addOutputState(SampleState(ourIdentity), SampleContract.CONTRACT_NAME)
+                .addCommand(SampleContract.Commands.Issue(), ourIdentity.owningKey)
+                .addReferenceState(ReferencedStateAndRef(ourMembership))
+
+        builder.verify(serviceHub)
+        val stx = serviceHub.signInitialTransaction(builder)
+        return subFlow(FinalityFlow(stx, listOf()))
+    }
+}
+
+@InitiatingFlow
+class TransferAssetFlow(val sampleState : StateAndRef<SampleState>,
+                              val bno : Party,
+                              val partyToTransferTo : Party) : FlowLogic<SignedTransaction>() {
+    @Suspendable
+    override fun call() : SignedTransaction {
+        if (ourIdentity != sampleState.state.data.owner)
+            throw FlowException("Only owner of the state can transfer it")
+        if (ourIdentity == partyToTransferTo)
+            throw FlowException("sender and recipient should be different parties")
+        // for the sake of demo we are just taking the first available notary
+        val notary = serviceHub.networkMapCache.notaryIdentities.single()
+
+        // get the membership from the node's membership cache
+        val membershipsCacheHolderService = serviceHub.cordaService(MembershipsCacheHolder::class.java)
+        val ourMembership = membershipsCacheHolderService.cache.getMembership(bno, ourIdentity)
```

Comment:
Would make sense to use the same API across the whole app. I.e. either `MembershipsCacheHolder` or `GetMembershipFlow`

Reply:
Used GetMembershipFlow everywhere.
Done.